src/interactive-window/interactiveWindowProvider.ts

```diff
@@ -273,6 +291,18 @@ export class InteractiveWindowProvider
         }
         return result;
     }
+    private _updateWindowCache() {
+        const windowCache = this._windows.map(
+            (iw) =>
+                ({
+                    owner: iw.owner,
+                    mode: iw.mode,
+                    uri: iw.notebookUri,
+                    inputBoxUri: iw.inputUri
+                } as IInteractiveWindowCache)
+        );
+        this.workspaceMemento.update(InteractiveWindowCacheKey, windowCache).then(noop, noop);
```

Comment:
I do'nt tihnk we can store Uris in this manner in the memento.
If you try to get the value out it will not be de-serialized into Uri class objects, but will be some funky JSON object.
I think its safer to store uri.toString() or the like, and then when reading this  value from cache we should probably use Uri.parse.

This way there's no assumption that things are serialized or anyone trying to use this in the future knows that its not a Uri object, but a plain string that needs to be parsed.

Also, i think we should store the controllerId (or kernelConnection.id = both are the same) at this point.

Reply:
> This way there's no assumption that things are serialized or anyone trying to use this in the future knows that its not a Uri object

Agree this is probably better to check if they need to perform `Uri.from`.